### ISSUE TITLE
New command - Open repo in terminal / CLI / bash

### ISSUE THREAD

**Issue by JeffMv** [sublimehq/sublime_merge#636]
**Problem description**

I would love to have a shortcut for opening the current repository in a new terminal window, just like [GitHub Desktop](https://desktop.github.com) application, which uses the `Ctrl` + ` \` ` shortcut along with a menu item.

Currently one has to open the Terminal app, open the repository in the Finder and `cd` + drag-n-drop the repo into the Terminal window. Or remember the repo's path by ❤️ .

**Preferred solution**

Adding a menu item (for instance in the Repository menu, next to "Reveal in Finder/Explorer") along with a command entry.

**Alternatives**

Note that I use `bash` on my Unix systems, but some people might want this feature to be able to open their favorite shell `sh`/`csh`/.... in which case you may make the preferred shell to be editable in settings. ... But then the way it might become OS-dependant (?) depending on how you choose to implement it.

**Screenshots**

-

**Comment by srbs** [comment]
related: #186

**Comment by JeffMv** [comment]
Thank you @srbs . Somehow I did not get the search result when searching for *opened issues* about terminal / CLI...


*For those stumbling upon this issue later on*

**TLDR** of other issues (like #186 and #201):
it's very difficult to support terminals across multiple platforms, and opening a specific folder would require a lot of hacks. On top of that, people would like options (for instance to open a new *tab* in Terminal instead of a new Terminal window)... A real non trivial mess.

**Workaround**
Hence, I found out that I was satisfied with only adding to the `Command Palette` the command to **open the current repo's directory in Finder/Windows Explorer**.

**How to**
And to do so, just add the file `Default.sublime-commands` in `Packages/User/`. (You can open the folder by clicking in `Preferences > Browse Packages`)
Here is mine. I also included 3 commands I found useful for Preferences.
```javascript
// Commands added to the command palette
// See https://github.com/sublimehq/sublime_merge/issues/186
// for some interesting commands
[
    // Adds the command "Open Repository Directory In File Explorer"
    // which will open the repo in the Finder/Windows Explorer/...
    {
        "caption": "Open Repository Directory In File Explorer",
        "command": "open_dir",
        "args": { "dir": "$working_dir" }
    },


    // Features I find handy to have, like in Sublime Text
    {
        "caption": "Preferences: Edit Settings…",
        "command": "open_in_editor",
        "args":
        {
            "path": "${packages}/User/Preferences.sublime-settings",
            "default": "// Settings in here override those in \"Default/Preferences.sublime-settings\",\n// and are overridden in turn by syntax-specific settings.\n{\n\n}\n"
        },
    },
    {
        "caption": "Preferences: Edit Commit Message Settings…",
        "command": "open_in_editor",
        "args":
        {
            "path": "${packages}/User/Commit Message.sublime-settings",
            "default": "// Settings in here override those in \"Default/Commit Message.sublime-settings\"\n{\n\n}\n"
        },
    },
    {
        "caption": "Preferences: Browse Packages…",
        "command": "open_dir",
        "args": { "dir": "$packages" },
    }
]
```

PS: I got the workaround I really wanted so I'm closing this issue
